build.gradle

```diff
@@ -0,0 +1,284 @@
+        entry 'okhttp'
+        entry 'mockwebserver'
+      }
+      // TODO provided
+      dependency 'xml-apis:xml-apis:1.4.01'
+      dependency 'org.sonarsource.orchestrator:sonar-orchestrator:3.16.0.1420'
+    }
+  }
+
+  apply plugin: 'java'
+  sourceCompatibility = 1.8
+  tasks.withType(JavaCompile) {
+    options.encoding = 'UTF-8'
+  }
+
+  def tempDir = file("$buildDir/temp")
+  test.doFirst {
+    // TODO note that due to this any change in this file (even addition of blank line) is considered as modification of input for tests
+    delete tempDir
+    tempDir.mkdirs()
+  }
+  test {
+    systemProperty 'java.io.tmpdir', tempDir
+    systemProperty 'java.awt.headless', true 
+    maxHeapSize = "512m"
+    jacoco.enabled = project.hasProperty('jacocoEnabled')
+    testLogging {
+      exceptionFormat 'full'
+      events "skipped", "failed"
+    }
+  }
+
+  apply plugin: 'com.google.protobuf'
+  sourceSets.main.proto.srcDir 'src/main/protobuf'  // in addition to the default 'src/main/proto'
+  sourceSets.test.proto.srcDir 'src/test/protobuf'  // in addition to the default 'src/test/proto'
+  protobuf {
+    protoc {
+      artifact = "com.google.protobuf:protoc:${protobufVersion}"
+    }
+  }
+
+  if (release) {
+    jar {
+      manifest {
+        attributes(
+          'Version': "${version}",
+          'Implementation-Build': 'git rev-parse HEAD'.execute().text.trim(),
+          'Build-Time': new Date().format("yyyy-MM-dd'T'HH:mm:ssZ")
+        )
+      }
+    }
+  }
+
+  apply plugin: 'idea'
+  idea {
+    module {
+      sourceDirs += file("${protobuf.generatedFilesBaseDir}/main/java")
+      testSourceDirs += file("${protobuf.generatedFilesBaseDir}/test/java")
+    }
+  }
+}
+
+configurations {
+  archives
+}
+
+apply plugin: 'maven-publish'
+// TODO requires review
+artifactory {
+  clientConfig.setIncludeEnvVars(true)
+  clientConfig.setEnvVarsExcludePatterns('*password*,*PASSWORD*,*secret*,*MAVEN_CMD_LINE_ARGS*,sun.java.command,*token*,*TOKEN*,*LOGIN*,*login*')
+  contextUrl = System.getenv('ARTIFACTORY_URL')
+  publish {
+    repository {
+      repoKey = System.getenv('ARTIFACTORY_DEPLOY_REPO')
+      username = System.getenv('ARTIFACTORY_DEPLOY_USERNAME')
+      password = System.getenv('ARTIFACTORY_DEPLOY_PASSWORD')
+    }
+    defaults {
+      properties = [
+        'vcs.revision': (System.getenv('GIT_COMMIT') ?: System.getenv('TRAVIS_COMMIT')),
+        'vcs.branch': (System.getenv('GIT_BRANCH') ?: System.getenv('TRAVIS_BRANCH')),
+        'build.name': 'sonarqube',
+        'build.number': (System.getenv('BUILD_ID') ?: System.getenv('TRAVIS_BUILD_NUMBER'))
+      ]
+      publications('mavenJava')
+      publishPom = true // Publish generated POM files to Artifactory (true by default)
+      publishIvy = false // Publish generated Ivy descriptor files to Artifactory (true by default)
+    }
+  }
+  clientConfig.info.setBuildNumber(System.getenv('TRAVIS_BUILD_NUMBER'))
+  // Give artifacts to the delivery process
```

Comment:
The comment could be more detailed, for example "Artifacts to be deployed to https://sonarsource.bintray.com on releases".